IdentityServer/v7/docs/content/fundamentals/keys/automatic_key_management.md

```diff
@@ -101,6 +101,14 @@ multiple keys, algorithms, and if those keys should additionally get wrapped in
 an X.509 certificate. Automatic key management will create and rotate keys for
 each signing algorithm you specify.
 
+{{% notice info %}}
+
+When using a *X.509 certificates*, the certificate may have an expiration date, but IdentityServer does
+not use this data to validate the certificate and throw an exception. If a certificate has expired then you
+must decide whether to continue using it or replace it with a new certificate.
```

Comment:
To stop using the certificate would also be somewhat no-trivial because you would probably want to rotate the secret, and there isn't an easy way to tell automatic key management that the secret is to be rotated. Probably you would have to convert the secret from an automatically managed secret to a manual secret - basically pull the data out of your database and then add it to IdentityServer as a manual key. And then you would want to go through the key rotation process manually (similar to what we document [here](https://docs.duendesoftware.com/identityserver/v7/fundamentals/keys/migration/), when going from manual key management to automatic key management).

Reply:
Agreed, but that's something the user has to decide for themselves, so it's not something we would want to elaborate on unless we have a clear guidance.